src/components/icon/icon.pw.tsx

```diff
@@ -0,0 +1,538 @@
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+      await expect(tooltip).toHaveCSS("background-color", tooltipBgColor);
+    });
+  });
+
+  colorData.forEach((tooltipFontColor) => {
+    test(`should check tooltip font color as ${tooltipFontColor}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconTooltipComponent tooltipFontColor={tooltipFontColor} />);
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+      await expect(tooltip).toHaveCSS("color", tooltipFontColor);
+    });
+  });
+
+  testData.forEach((tooltipId) => {
+    test(`should check tooltipId as ${tooltipId}`, async ({ mount, page }) => {
+      await mount(<IconTooltipComponent tooltipId={tooltipId} />);
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+      await expect(tooltip).toHaveId(tooltipId);
+    });
+  });
+
+  (["error", "add", "admin", "alert"] as IconProps["type"][]).forEach(
+    (iconType) => {
+      test(`should check iconType as ${iconType}`, async ({ mount, page }) => {
+        await mount(<IconComponent type={iconType} />);
+
+        const iconLocator = await page.getByTestId("icon");
+        await iconLocator.click();
+        await expect(iconLocator).toHaveAttribute("type", iconType);
+      });
+    }
+  );
+
+  ([
+    [true, "rgba(0, 0, 0, 0.3)"],
+    [false, "rgba(0, 0, 0, 0.9)"],
+  ] as [IconProps["disabled"], string][]).forEach(([boolVal, color]) => {
+    test(`should check icon color when disabled prop is set as ${boolVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconComponent disabled={boolVal} />);
+
+      const iconLocator = await page.getByTestId("icon");
+      await expect(iconLocator).toHaveCSS("color", color);
+    });
+  });
+
+  colorData.forEach((backgroundColor) => {
+    test(`should check background color as ${backgroundColor}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconComponent bg={backgroundColor} />);
+
+      const iconLocator = await page.getByTestId("icon");
+      await expect(iconLocator).toHaveCSS("background-color", backgroundColor);
+    });
+  });
+
+  colorData.forEach((iconColor) => {
+    test(`should check icon color as ${iconColor}`, async ({ mount, page }) => {
+      await mount(<IconComponent color={iconColor} />);
+
+      const iconLocator = await page.getByTestId("icon");
+      await expect(iconLocator).toHaveCSS("color", iconColor);
+    });
+  });
+
+  ([
+    [SIZE.SMALL, 24],
+    [SIZE.MEDIUM, 32],
+    [SIZE.LARGE, 40],
+    [SIZE.EXTRALARGE, 56],
+  ] as [IconProps["fontSize"], number][]).forEach(
+    ([fontSize, heightAndWidth]) => {
+      test(`should check height and width as ${heightAndWidth} when fontSize is set as ${fontSize}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<IconComponent fontSize={fontSize} />);
+
+        const iconLocator = await page.getByTestId("icon");
```

Comment:
await is not necessary here. This will work: 
`const iconLocator = page.getByTestId("icon");`